### ISSUE TITLE
nextCloudCmd throws error 'Sqlite exec statement error: 5 "database is locked"'

### ISSUE THREAD

**Issue by pb-businessacts** [nextcloud/desktop#3840]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
nextCloudCmd shoud start and work.

### Actual behaviour
I get the error 'Sqlite exec statement error: 5 "database is locked" (see screenshot) and somtimes after a long waitingtime the synchronisation starts.
![image](https://user-images.githubusercontent.com/91463049/134883079-d085575c-1aa8-4841-8854-e01189cb4199.png)

### Steps to reproduce
1. Start nexcloudCmd for initialize the clientside files
2. After initial sychronisation start nextcloudCmd again

My call:
"C:\Program Files (x86)\appdir\nextcloudcmd.exe" --trust --path "sellify_documents_dev/sellifyontoursync" --non-interactive  -u myuser -p mypassword --unsyncedfolders "C:\Program Files (x86)\IgnoreLists\default_ignore.txt" C:\sellify_documents https://myserver/nextcloud 

### Client configuration
Client version:
Nextcloud version 3.3.2stable-Win64 (build 20210826)
Git revision 50453d9d7e5b42749b86e7a7f0e6b61b69b79650
Using Qt 5.12.9, built against Qt 5.12.9
Using 'OpenSSL 1.1.1g  21 Apr 2020'
Running on Windows Server 2016 (10.0), x86_64

Operating system:
Windows Server 2019 Standard 64bit

OS language:
German

Installation path of client:

Storage backend (external storage):
Networkdrive

**Comment by FlexW** [comment]
Is your nextcloud client running? You probably need to stop it. The database gets locked exclusively by it.

**Comment by pb-businessacts** [comment]
No, nextCloudClient is not running on this machine.

**Comment by Senth** [comment]
I get the same problem, and with 10+ folders to sync, this means my sync takes approx 10min when I used selective sync.

**Comment by beardhatcode** [comment]
I also have this issue with:

```
Nextcloud version 3.3.6git
Git revision 6d3270dd251db42293233d5622a610ea172dabef
Using Qt 5.15.3, built against Qt 5.15.3
Using 'OpenSSL 1.1.1l  24 Aug 2021'
Running on NixOS 22.05 (Quokka), x86_64
```

**Comment by uwe347** [comment]
Same problem with:
```
$ nextcloudcmd --version
Nextcloud version 3.4.0-20211129.184724.c004db207-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.3 LTS, x86_64
```
I observed that this only happens together with option --unsyncedfolders. Logging output shows:
```
12-01 09:23:47:292 [ info nextcloud.sync.database ]:	sqlite3 version "3.31.1"
12-01 09:23:47:292 [ info nextcloud.sync.database ]:	sqlite3 locking_mode= "exclusive"
12-01 09:23:47:292 [ info nextcloud.sync.database ]:	sqlite3 journal_mode= "wal"
12-01 09:23:47:292 [ info nextcloud.sync.database ]:	sqlite3 synchronous= "NORMAL"
12-01 09:25:29:542 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "UPDATE metadata SET md5='_invalid_' WHERE (?1 == path OR (?1 > (path||'/') AND ?1 < (path||'0'))) AND type == 2;"
12-01 09:27:11:788 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
12-01 09:27:11:788 [ warning nextcloud.sync.database ]:	SQL error when deleting selective sync list ("Archiv/", "weitereDOTX/") "database is locked"
12-01 09:28:54:030 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
12-01 09:28:54:030 [ warning nextcloud.sync.database ]:	SQL error when inserting into selective sync 1 "Archiv/" "database is locked"
12-01 09:30:36:282 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
12-01 09:30:36:282 [ warning nextcloud.sync.database ]:	SQL error when inserting into selective sync 1 "weitereDOTX/" "database is locked"
12-01 09:30:36:282 [ warning nextcloud.sync.database ]:	ERROR committing to the database: "cannot commit transaction - SQL statements in progress"
```
After this, syncing resumes quite normally - but a change in my --unsyncedfolders -file is not obeyed. This is not surprising since the above "warning"-Message tells "SQL error when inserting into selective sync" (in that case i added the line "weitereDOTX" to my --unsyncedfolders -file, which before hat only one line: "Archiv"). So fixing this bug is not only necessary to avoid the "database locked" -delay (which takes about five minutes), but also to bring any change in  --unsyncedfolders -file to work.

**Comment by beardhatcode** [comment]
Problem still exists on 3.4.1 fa32c100148925092c9a320e6f8e2e0130b9cb6a
```
$ nextcloudcmd --version
Nextcloud version 3.4.1git
Git revision fa32c100148925092c9a320e6f8e2e0130b9cb6a
Using Qt 5.15.3, built against Qt 5.15.3
Using 'OpenSSL 1.1.1l  24 Aug 2021'
Running on NixOS 22.05 (Quokka), x86_64
```

**Comment by Tachi107** [comment]
> This bug report is getting automatically closed due to no answer since the issue has been staled. Thank you!

That's wrong, please reopen this issue as it is not fixed

**Comment by beardhatcode** [comment]
I get the error for `DELETE FROM selectivesync WHERE type == ?1`.

```log
01-25 11:10:06:228 [ info nextcloud.sync.accessmanager ]:       2 "" "https://HOST/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "b8962a4e-2aa8-4305-b0a4-8608b739d654"
01-25 11:10:06:228 [ info nextcloud.sync.networkjob ]:  OCC::JsonApiJob created for "https://HOST" + "ocs/v1.php/cloud/capabilities" ""
01-25 11:10:09:876 [ info nextcloud.sync.networkjob.jsonapi ]:  JsonApiJob of QUrl("https://HOST/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "OK"
01-25 11:10:09:876 [ debug default ]    [ main(int, char**)::<lambda ]: Server capabilities QJsonObject(...REDACTED...)
01-25 11:10:09:876 [ info nextcloud.sync.accessmanager ]:       2 "" "https://HOST/ocs/v1.php/cloud/user?format=json" has X-Request-ID "REDACTED"
01-25 11:10:09:877 [ info nextcloud.sync.networkjob ]:  OCC::JsonApiJob created for "https://HOST" + "ocs/v1.php/cloud/user" ""
01-25 11:10:11:588 [ info nextcloud.sync.networkjob.jsonapi ]:  JsonApiJob of QUrl("https://HOST/ocs/v1.php/cloud/user?format=json") FINISHED WITH STATUS "OK"
01-25 11:10:11:733 [ info nextcloud.sync.database ]:    sqlite3 version "3.37.0"
01-25 11:10:11:733 [ info nextcloud.sync.database ]:    sqlite3 locking_mode= "exclusive"
01-25 11:10:11:733 [ info nextcloud.sync.database ]:    sqlite3 journal_mode= "wal"
01-25 11:10:11:733 [ info nextcloud.sync.database ]:    sqlite3 synchronous= "NORMAL"
01-25 11:11:53:926 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
01-25 11:11:53:927 [ warning nextcloud.sync.database ]: SQL error when deleting selective sync list ("InstantUpload/") "database is locked"
01-25 11:13:36:152 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
01-25 11:13:36:152 [ warning nextcloud.sync.database ]: SQL error when inserting into selective sync 1 "InstantUpload/" "database is locked"
01-25 11:13:36:152 [ warning nextcloud.sync.database ]: ERROR committing to the database: "cannot commit transaction - SQL statements in progress"
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      There are 1175386542080 bytes available at "/home/beardhatcode/Nextcloud/"
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      Sync with existing sync journal
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      "Using Qt 5.15.3 SSL library OpenSSL 1.1.1l  24 Aug 2021 on NixOS 22.05 (Quokka)"
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      Using Selective Sync
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      #### Discovery start ####################################################
01-25 11:13:36:160 [ info nextcloud.sync.engine ]:      Server ""
01-25 11:13:36:167 [ info sync.discovery ]:     STARTING "" OCC::ProcessDirectoryJob::NormalQuery "" OCC::ProcessDirectoryJob::NormalQuery
...
```

With 
```
[nix-shell:~]$ nextcloudcmd --version
Nextcloud version 3.4.1git
Git revision fa32c100148925092c9a320e6f8e2e0130b9cb6a
Using Qt 5.15.3, built against Qt 5.15.3
Using 'OpenSSL 1.1.1l  24 Aug 2021'
Running on NixOS 22.05 (Quokka), x86_64
```

**Comment by brendan-pike** [comment]
Also effecting me. 
02-07 19:36:10:571 [ warning nextcloud.sync.database ]:	ERROR committing to the database:  "database is locked"

ps.  auto closing tickets is such a false economy.

**Comment by Adspectus** [comment]
I have the same problem with:
```
Nextcloud version 3.4.2-20220209.150345.b0791e51c-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.4 LTS, x86_64
```

**Comment by Adspectus** [comment]
I noticed that the error happen, when the local dir is on a NFS4 mounted directory from another machine. On the local filesystem, all went well.

**Comment by Tachi107** [comment]
Again...

**Comment by STonEDSTonE** [comment]
I do not understand why this issue does get closed automatically all the time even if there clearly are people being affected.

I also am afffected, running Nextcloud 23.0.4 running on Ubuntu 20.04.4 LTS, using mysql 8.0.28 as database and apache 2.4.53 as webserver. For sync i use nextcloudcmd version 3.4.4.
When i run nextcloudcmd, it prints out some sqlite3 Infos, takes several minutes (see timestamps) while printing out SQL Errors and then starts sync without much hassle.

`nextcloudcmd -u "$USER1" -p "$PW1" --unsyncedfolders "$PATH"/sunsyncedfolders_"$USER1".lst "$DIR" "$URL"`

> 04-28 10:44:27:227 [ info nextcloud.sync.accessmanager ]:	2 "" "URL/ocs/v1.php/cloud/user?format=json" has X-Request-ID "763105b6-2d8b-4c86-886a-f0d3ce91a104"
> 04-28 10:44:27:227 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "URL" + "ocs/v1.php/cloud/user" ""
> 04-28 10:44:38:657 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("URL/ocs/v1.php/cloud/user?format=json") FINISHED WITH STATUS "OK"
> 04-28 10:44:38:752 [ info nextcloud.sync.database ]:	sqlite3 version "3.31.1"
> 04-28 10:44:38:752 [ info nextcloud.sync.database ]:	sqlite3 locking_mode= "exclusive"
> 04-28 10:44:38:752 [ info nextcloud.sync.database ]:	sqlite3 journal_mode= "wal"
> 04-28 10:44:38:752 [ info nextcloud.sync.database ]:	sqlite3 synchronous= "NORMAL"
> 04-28 10:46:20:918 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
> 04-28 10:46:20:918 [ warning nextcloud.sync.database ]:	SQL error when deleting selective sync list ("Temp/", "Talk/") "database is locked"
> 04-28 10:48:03:086 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
> 04-28 10:48:03:086 [ warning nextcloud.sync.database ]:	SQL error when inserting into selective sync 1 "Temp/" "database is locked"
> 04-28 10:49:45:258 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
> 04-28 10:49:45:258 [ warning nextcloud.sync.database ]:	SQL error when inserting into selective sync 1 "Talk/" "database is locked"
> 04-28 10:49:45:258 [ warning nextcloud.sync.database ]:	ERROR committing to the database: "cannot commit transaction - SQL statements in progress"
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	There are 1790521798656 bytes available at "PATH"
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	Sync with existing sync journal
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	"Using Qt 5.12.8 SSL library OpenSSL 1.1.1f  31 Mar 2020 on Ubuntu 20.04.4 LTS"
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	Using Selective Sync
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	#### Discovery start ####################################################
> 04-28 10:49:45:267 [ info nextcloud.sync.engine ]:	Server "" 
> 04-28 10:49:45:267 [ info sync.discovery ]:	STARTING "" OCC::ProcessDirectoryJob::NormalQuery "" OCC::ProcessDirectoryJob::NormalQuery
> 04-28 10:49:45:267 [ info nextcloud.sync.accessmanager ]:	6 "PROPFIND" "URL/remote.php/dav/files/USER/" has X-Request-ID "7a617833-6bd3-42b9-8b50-8c6d4acfe085"
> 04-28 10:49:45:268 [ info nextcloud.sync.networkjob ]:	OCC::LsColJob created for "URL" + "/" "OCC::DiscoverySingleDirectoryJob"
> 04-28 10:49:46:021 [ info nextcloud.sync.networkjob.lscol ]:	LSCOL of QUrl("URL/remote.php/dav/files/USER/") FINISHED WITH STATUS "OK"
> 04-28 10:49:46:022 [ warning default ]:	System exclude list file could not be read: "PATH/.sync-exclude.lst"
> 04-28 10:49:46:022 [ info sync.discovery ]:	Processing "FOLDER" | valid: true/true/true | mtime: 1651072305/1650819234/1651072305 | size: 0/4096/0 | etag: "62695d31d7e79"//"62695d31d7e79" | checksum: ""//"" | perm: "DNVCKR"//"DNVCKR" | fileid: "00000429oc3h0w4dc64t"//"00000429oc3h0w4dc64t" | inode: 74973190/74973190/ | type: CSyncEnums::ItemTypeDirectory/CSyncEnums::ItemTypeDirectory/CSyncEnums::ItemTypeDirectory | e2ee: false/false | e2eeMangledName: ""/""

I also have several nextcloudcmd syncs running per hour so this takes up quite some time, so it would be good to see this fixed.
Can someone reopen this please?

**Comment by Senth** [comment]
I've lost hope on this being fixed...

6 weeks of stale is way too short amount of time. Feels more like they want to keep bug reports to a minimum as people don't want to recreate the bug again. The 👍 can't be used either. 12 months would be more realistic.

Almost thinking of creating a bot that recreate the issue directly after its closed 🤣 But I'll live with not using selective sync and use the 50GB extra storage for now.

**Comment by STonEDSTonE** [comment]
At least it would be nice to know why this gets closed.
Or even an explaination why this bug is so hard to fix or not fixable at all.
In https://github.com/nextcloud/desktop/issues/3222 there is sth. about `immutable=1` that may or may not help, I can not tell and do not understand the explaination :/

**Comment by klausman** [comment]
Here to chime in that this has started happening to me recently (likely due to a distro upgrade to bookworm). 

In my case, the client is running in an LXC container. Combined with some reports of this happening on NFS and vut not on that users local FSes, I suspect some locking primitives used for WAL by sqlite here just don't work in those case. But with zero support (or interest, it seems) from the devs, it's hard to say.

**Comment by kceiw** [comment]
This happens to me too.
nextcloudcmd version is 3.5.1

The logic used by the bot to close the bug doesn't really make sense. From most of comment I can tell people keep reporting and reopening the bug.  I may miss it but I don't see the maintainer to interact much. The comment list becomes the tug and turf between users and the bot to reopen and close the bug.  I'll agree to that the bug is stale when

**Comment by perry-mitchell** [comment]
Also getting this error, but without unsyncedfolders.. just a selective directory sync (remote).

**Comment by aHcVolle** [comment]
Same problem here

**Comment by elfen20** [comment]
I had the same problem when using `--unsyncedfolders`. The nextcloud command seems to open a second database connection so set the unsyncedfolders values. This fails because of `sqlite3 locking_mode= "exclusive"`and `sqlite3 journal_mode= "wal"`.
I found out a workaround is to reset the journal mode to delete before starting the nextcloud client.
I have a entrypoint.sh inside a docker container, it searches for a db file in the folder nextcloud and changes the journal mode:
```
#!/bin/bash
# entrypoint.sh
echo "Resetting DB Journal Mode..."
dbfile=( /nextcloud/.*.db )
/usr/bin/sqlite3 $dbfile 'PRAGMA journal_mode = delete;'

echo "Running nextcloud sync.."
nextcloudcmd --unsyncedfolders /config/unsyncedfolders /nextcloud https://$USER:$PWD@server.tld
```

**Comment by mayonezo** [comment]
Same issue for me. Additionally, I would really appreciate documentation or an example of an unsyncedfolders file.

**Comment by elfen20** [comment]
The unsynced folders file is just a text file with one folder per line.
```
Upload
Media
Folder/Subfolder
```

Example:
[https://github.com/elfen20/nextcloud-docker](url)

**Comment by mayonezo** [comment]
Your link does not work for me.

I have also seen files with a trailing slash:

Upload/
Media/
Folder/Subfolder/

**Comment by elfen20** [comment]
Oh sorry, the Link was wrong.
A trailing slash makes no difference.

For example if you have online
`Folder/Sub1/Sub2`

then with the line `Folder/Sub1`or `Folder/Sub1/` in the unsyncedfolders file
you will get all items in `Folder` except `Sub1` locally.

**Comment by mayonezo** [comment]
So is there a way to reopen this bug?

**Comment by beardhatcode** [comment]
@elfen20 , is that the locking mode that is set here?

https://github.com/nextcloud/desktop/blob/8fd4519d35232841c35dc17a22d51c7fcba9db82/src/common/syncjournaldb.cpp#L318-L326

**Comment by elfen20** [comment]
I don't know about the underlying problem of the combination of locking mode, journaling mode and file access, but there is already a workaround implemented for windows and mac. I think i run into the same problem with the mounted storage in my container. 

https://github.com/nextcloud/desktop/blob/8fd4519d35232841c35dc17a22d51c7fcba9db82/src/common/syncjournaldb.cpp#L82-L102

**Comment by elfen20** [comment]
@beardhatcode , thanks for the code snippet! I could remove the manual setting of the journaling mode and use the environment variable instead. Setting
`OWNCLOUD_SQLITE_JOURNAL_MODE=DELETE`
fixed it.

**Comment by mayonezo** [comment]
@beardhatcode Could you reopen this issue?

**Comment by mayonezo** [comment]
> @beardhatcode , thanks for the code snippet! I could remove the manual setting of the journaling mode and use the environment variable instead. Setting `OWNCLOUD_SQLITE_JOURNAL_MODE=DELETE` fixed it.

Is that a fix that can be applied in a PR?

**Comment by beardhatcode** [comment]
It might be, I'll reopen the issue

**Comment by mayonezo** [comment]
Could someone add the approved label, so github-actions won't close it again?

**Comment by mayonezo** [comment]
@mgallien Could you approve this issue, please?

**Comment by uwe347** [comment]
I tested elfen20' workaround
`export OWNCLOUD_SQLITE_JOURNAL_MODE=DELETE`
but without success. The (as  I think) relevant messages have changed from
```
01-20 05:12:40:199 [ info nextcloud.sync.database ]:    sqlite3 version "3.31.1"
01-20 05:12:40:199 [ info nextcloud.sync.database ]:    sqlite3 locking_mode= "exclusive"
01-20 05:12:40:199 [ info nextcloud.sync.database ]:    sqlite3 journal_mode= "wal"
01-20 05:12:40:199 [ info nextcloud.sync.database ]:    sqlite3 synchronous= "NORMAL"
01-20 05:14:22:446 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
01-20 05:14:22:447 [ warning nextcloud.sync.database ]: SQL error when deleting selective sync list ("archiv/") "database is locked"
01-20 05:16:04:695 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
01-20 05:16:04:696 [ warning nextcloud.sync.database ]: SQL error when inserting into selective sync 1 "archiv/" "database is locked"
01-20 05:16:04:696 [ warning nextcloud.sync.database ]: ERROR committing to the database: "cannot commit transaction - SQL statements in progress"

```
to
```
01-20 05:19:54:469 [ info nextcloud.sync.database ]:    sqlite3 version "3.31.1"
01-20 05:19:54:469 [ info nextcloud.sync.database ]:    sqlite3 locking_mode= "exclusive"
01-20 05:19:56:474 [ warning nextcloud.sync.database.sql ]:     Sqlite step statement error: 5 "database is locked" in "PRAGMA journal_mode=DELETE;"
01-20 05:19:56:474 [ info nextcloud.sync.database ]:    sqlite3 journal_mode= ""
01-20 05:19:56:474 [ info nextcloud.sync.database ]:    sqlite3 synchronous= "FULL"
01-20 05:21:38:731 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
01-20 05:21:38:731 [ warning nextcloud.sync.database ]: SQL error when deleting selective sync list ("archiv/") "database is locked"
01-20 05:23:20:982 [ warning nextcloud.sync.database.sql ]:     Sqlite exec statement error: 5 "database is locked" in "INSERT INTO selectivesync VALUES (?1, ?2)"
01-20 05:23:20:982 [ warning nextcloud.sync.database ]: SQL error when inserting into selective sync 1 "archiv/" "database is locked"
01-20 05:23:20:982 [ warning nextcloud.sync.database ]: ERROR committing to the database: "cannot commit transaction - SQL statements in progress"

```
Journal mode changed from "wal" to "", but nevertheless the database still is locked and this takes 2 minutes 42 sec (as you can see in the above log between  05:19:56:474 and 05:21:38:731). 

My version is:
```
$ nextcloudcmd --version
Nextcloud version 3.5.4-20220806.084713.fea986309-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.5 LTS, x86_64
```

**Comment by elfen20** [comment]
I've never seen `journal_mode= ""`...
SQLite changes the the journalmode to "wal" when opening the db, but i think returning to "delete" can not be done automatically.  
See https://www.sqlite.org/wal.html  
I have changed the journal mode one time before running the nextcloud client with  
`/usr/bin/sqlite3 [PATH_TO_DB_FILE] 'PRAGMA journal_mode = delete;'`  
after that setting `export OWNCLOUD_SQLITE_JOURNAL_MODE=DELETE` keeps the journaling mode at "delete".
You can also try to remove/rename the database so nextcloud will create  a new one and sets the journaling mode correct.

**Comment by uwe347** [comment]
Yes, that's exactly what I did, based on your hint: `OWNCLOUD_SQLITE_JOURNAL_MODE=DELETE`
The result is a full synchronization of all files. But as you can see above: the error "database is locked" is still there - which costs time and inhibits the correct function of  option --unsyncedfolders

**Comment by mgallien** [comment]
@uwe347 @mayonezo 
not sure I have any time for this issue but could you give me a list of reproduction steps ?
that is needed to add the approved label and would be a prerequisite to work on a fix
those reproduction steps would then be created as a test case and we would be sure that this is properly fixed

**Comment by mayonezo** [comment]
The following command gives me the error:

nextcloudcmd -n --unsyncedfolders ./unsyncedfolders.lst /home/user/Nextcloud_Backup https://domain.com/nextcloud/

When I don't pass the --unsynchedfolders switch, it works without issues.

**Comment by meusburgersystems** [comment]
Same/similar Problem here.
OS: Windows 10
NextCloud Client: 3.7.3

Steps to reproduce:
Run a Sync with --unsyncedfolders Parameter --> 1st run OK
2nd run takes ages and fails at all with "Database locked" errors and ending ip in a retry loop:

```
02-26 10:54:40:978 [ warning nextcloud.sync.database.sql ]:	Sqlite exec statement error: 5 "database is locked" in "DELETE FROM selectivesync WHERE type == ?1"
02-26 10:54:40:978 [ warning nextcloud.sync.database ]:	SQL error when deleting selective sync list ("********/") "database is locked"
```

**Comment by mayonezo** [comment]
@mgallien Could you add the approved label, please?

**Comment by mayonezo** [comment]
Reacting to github-actions auto-close: Bug is still there.

**Comment by kceiw** [comment]
Add a comment so that it's not closed by bot. We wait for the developers.

**Comment by mayonezo** [comment]
@mgallien Could you add the approved label, please?

**Comment by mayonezo** [comment]
@mgallien Could you add the approved label, please?

**Comment by mayonezo** [comment]
@claucambra Could you add the approved label, please?

**Comment by mayonezo** [comment]
@mgallien @claucambra Could you please reopen and add the approved label to this issue?

**Comment by patrickunderscore** [comment]
I had the same issue on one of my linux clients after migration from owncloud (auto import parameters)...
After opening client parameters - and waiting a lot that the window appears - I saw that I had 2 times the same root sync folder, one fully synchronized, other still active. I deleted the last, and now all works fine !
Hope this help.

**Comment by tgwaste** [comment]
For God's sake.

--unsyncedfolders needs to be fixed, not --exclude (which I assume sync-exclude.lst is for).

This is STILL BROKE!!  If you're not going to support --unsyncedfolders, then freaking remove the option.

In my unsyncedfolders.exclude file I have:
folder1/

Using the above test for linux the exact same thing thats been happening for YEARS is still happening.
Please fix this already!

**Comment by mayonezo** [comment]
@camilasan Please reopen the issue. I don't think we are able to.

**Comment by mayonezo** [comment]
Again: @camilasan @mgallien @claucambra Could you please add the accepted label in order for this issue not to auto-stale?

**Comment by NeXX451** [comment]
Any update on this? This issue still persists.

**Comment by theFork** [comment]
Also occurs to me on 3.11.0git

**Comment by DimanNe** [comment]
Getting this error when trying to import tasks/calendar

**Comment by joshtrichards** [comment]
Duplicate of #3422
